File: typescript/spell-name.ts
import readline from "readline";

const rl = readline.createInterface({
	input: process.stdin,
	output: process.stdout,
});

rl.question("\nWhat is your name : ", (nameInput) => {
	let isLoop = true;
	let agreementInput = "";

	const askAgreement = () => {
		rl.question("\nCan I spell your name?\nEnter 'y' or 'n'. : ", (answer) => {
			agreementInput = answer.trim();

			if (!agreementInput) {
				askAgreement();
			} else {
				isLoop = false;
				if (agreementInput === "y") {
					for (let item of nameInput) {
						console.log(item);
					}
					console.log(`Here is your name spell, ${nameInput.trim()}!`);
				} else {
					console.log("Exiting the program...");
				}
				rl.close();
			}
		});
	};

	askAgreement();
});

File: rust/spell_name.rs
use std::io::{self, Write};

fn main() {
    let mut name_input = String::new();
    print!("\nWhat is your name : ");
    io::stdout().flush().unwrap();
    io::stdin()
        .read_line(&mut name_input)
        .expect("Failed to read input");

    let mut is_loop = true;
    let mut agreement_input = String::new();

    while is_loop {
        print!("\nCan I spell your name?");
        io::stdout().flush().unwrap();
        print!("\nEnter 'y' or 'n'. : ");
        io::stdout().flush().unwrap();

        io::stdin()
            .read_line(&mut agreement_input)
            .expect("Failed to read input");

        agreement_input = agreement_input.trim().to_string();

        if !agreement_input.is_empty() {
            is_loop = false;
        }
    }

    if agreement_input == "y" {
        for item in name_input.chars() {
            println!("{}", item);
        }
        println!("Here is your name spell, {}!", name_input.trim());
    } else {
        println!("Exiting the program...")
    }
}
